apps/hpc-ftsadmin/src/app/components/flow-form/flow-form.tsx

```diff
@@ -0,0 +1,2295 @@
+      <Box sx={tw`flex items-center justify-between`}>
+        <h2>{title}</h2>
+        {closeButtonAction && (
+          <MdClose
+            onClick={closeButtonAction}
+            style={{ cursor: 'pointer', fontSize: '25px' }}
+          />
+        )}
+      </Box>
+      {children}
+      {closeButtonAction && (
+        <Box sx={tw`pt-4 flex justify-end`}>
+          <C.Button
+            color="secondary"
+            onClick={closeButtonAction}
+            text={t.t(lang, (s) => s.components.reportingDetail.removeButton)}
+          />
+        </Box>
+      )}
+    </FormGroupPaper>
+  );
+};
+
+const BehaviorLabel = ({
+  values,
+  name,
+}: {
+  values: FlowFormType;
+  name: FlowFormFlowObjectKey;
+}) => {
+  if (values[name].length <= 1) {
+    return null;
+  }
+  const lang = getContext().lang;
+  const label = SHARED_FIELDS.has(name) ? 'shared' : 'overlap';
+
+  const { tooltip, label: chipLabel } = t.get(
+    lang,
+    (s) => s.components.flowForm.behaviorLabel[label]
+  );
+  return (
+    <Tooltip title={tooltip}>
+      <Chip label={chipLabel} size="small" sx={tw`ms-4`} color="primary" />
+    </Tooltip>
+  );
+};
+
+const FlowAmountButton = ({
+  amountUSD,
+  amountOriginalCurrency,
+  exchangeRate,
+  setFieldValue,
+  disabled,
+}: {
+  amountUSD: FlowFormType['amountUSD'];
+  amountOriginalCurrency: FlowFormType['amountOriginalCurrency'];
+  exchangeRate: FlowFormType['exchangeRate'];
+  setFieldValue: FormikHelpers<FlowFormType>['setFieldValue'];
+  disabled?: boolean;
+}) => {
+  const { lang } = getContext();
+
+  if (disabled) {
+    return;
+  }
+  const amountUSDInt = currencyToInteger(amountUSD);
+  const amountOriginalCurrencyInt = currencyToInteger(amountOriginalCurrency);
+  const exchangeRateFloat = parseFloat(exchangeRate);
+
+  if (amountUSDInt && !amountOriginalCurrencyInt && exchangeRateFloat) {
+    const buttonProps = {
+      onClick: () =>
+        setFieldValue(
+          'amountOriginalCurrency',
+          valueToInteger(amountUSDInt * exchangeRateFloat)
+        ),
+      text: t.t(
+        lang,
+        (s) => s.components.flowAmountButton.amountOriginalCurrency
+      ),
+    };
+    return <C.Button color="primary" {...buttonProps} className="text-end" />;
+  } else if (!amountUSDInt && amountOriginalCurrencyInt && exchangeRateFloat) {
+    const buttonProps = {
+      onClick: () =>
+        setFieldValue(
+          'amountUSD',
+          valueToInteger(amountOriginalCurrencyInt / exchangeRateFloat)
+        ),
+      text: t.t(lang, (s) => s.components.flowAmountButton.amountUSD),
+    };
+    return <C.Button color="primary" {...buttonProps} className="text-end" />;
+  }
+  const buttonProps = {
+    onClick: () =>
+      setFieldValue('exchangeRate', amountOriginalCurrencyInt / amountUSDInt),
+    text: t.t(lang, (s) => s.components.flowAmountButton.exchangeRate),
+  };
+  return <C.Button color="primary" {...buttonProps} className="text-end" />;
```

Comment:
To avoid returning the same thing 3 times, when this is about what gets assigned to `buttonProps`, you could do this:
```suggestion
  let buttonProps = {
    onClick: () =>
      setFieldValue('exchangeRate', amountOriginalCurrencyInt / amountUSDInt),
    text: t.t(lang, (s) => s.components.flowAmountButton.exchangeRate),
  };

  if (amountUSDInt && !amountOriginalCurrencyInt && exchangeRateFloat) {
    buttonProps = {
      onClick: () =>
        setFieldValue(
          'amountOriginalCurrency',
          valueToInteger(amountUSDInt * exchangeRateFloat)
        ),
      text: t.t(
        lang,
        (s) => s.components.flowAmountButton.amountOriginalCurrency
      ),
    };
  } else if (!amountUSDInt && amountOriginalCurrencyInt && exchangeRateFloat) {
    buttonProps = {
      onClick: () =>
        setFieldValue(
          'amountUSD',
          valueToInteger(amountOriginalCurrencyInt / exchangeRateFloat)
        ),
      text: t.t(lang, (s) => s.components.flowAmountButton.amountUSD),
    };
  }

  return <C.Button color="primary" {...buttonProps} className="text-end" />;
```